cmd/tailscaled/tailscaled.go

```diff
@@ -512,6 +512,27 @@ func getLocalBackend(ctx context.Context, logf logger.Logf, logID logid.PublicID
 		}
 	}
 	if socksListener != nil || httpProxyListener != nil {
+		dialer.UserDialCustomResolver = func(host string) (netip.Addr, error) {
```

Comment:
Please make this an exported function in net/dns and call it from here. We'll likely also use it from https://github.com/tailscale/tailscale/tree/main/tsnet